pageserver/compaction/src/identify_levels.rs

```diff
@@ -310,22 +349,45 @@ mod tests {
     }
 
     #[tokio::test]
-    async fn repro_identify_levels_bails_too_ealy_if_partitioned_keyspace_same_lsn() -> anyhow::Result<()> {
+    async fn test_disjoint_key_ranges_same_lsn() -> anyhow::Result<()> {
```

Comment:
I'm not sure this test case still makes sense, given that the original problem it was written to demonstrate turned out to be a red herring. identify_levels() doesn't pay any attention to the key ranges, so testing with partitioned keyspace isn't that different from the existing tests.

Reply:
b6d51d3b4ae8ddcb1e4a53854fb74f171ddc9310